src/ed25519.rs

```diff
@@ -431,15 +433,47 @@ impl<'d> Deserialize<'d> for Keypair {
             where
                 E: SerdeError,
             {
+                if bytes.len() != KEYPAIR_LENGTH {
```

Comment:
I'm unsure if this is ever possible, given serde's construction, but this check prevents panics if the provided slice is of the incorrect size.